qiskit/pulse/pulse_lib/sample_pulse.py

```diff
@@ -132,7 +133,9 @@ def draw(self, dt: float = 1,
                                           interactive=interactive)
 
     def __eq__(self, other: Pulse) -> bool:
-        return super().__eq__(other) and (self.samples == other.samples).all()
+        return super().__eq__(other) and np.allclose(self.samples, other.samples,
+                                                     rtol=other.epsilon,
+                                                     atol=self.epsilon)
```

Comment:
Since pulse samples are normalized to 1, perhaps the relative tolerance should be set to 0 and the absolute tolerance should be epsilon?